apps/angular/1-projection/src/app/ui/card/card.component.ts

```diff
@@ -1,61 +1,56 @@
-import { NgFor, NgIf } from '@angular/common';
-import { Component, Input } from '@angular/core';
-import { randStudent, randTeacher } from '../../data-access/fake-http.service';
-import { StudentStore } from '../../data-access/student.store';
-import { TeacherStore } from '../../data-access/teacher.store';
-import { CardType } from '../../model/card.model';
-import { ListItemComponent } from '../list-item/list-item.component';
+import { NgClass, NgForOf, NgTemplateOutlet } from '@angular/common';
+import {
+  ChangeDetectionStrategy,
+  Component,
+  ContentChild,
+  input,
+  output,
+} from '@angular/core';
+import { Id } from '../../model/id.model';
+import { CardItemDirective } from './card-item.directive';
 
 @Component({
   selector: 'app-card',
   template: `
     <div
       class="flex w-fit flex-col gap-3 rounded-md border-2 border-black p-4"
-      [class]="customClass">
-      <img
-        *ngIf="type === CardType.TEACHER"
-        src="assets/img/teacher.png"
-        width="200px" />
-      <img
-        *ngIf="type === CardType.STUDENT"
-        src="assets/img/student.webp"
-        width="200px" />
+      [ngClass]="customClass()">
```

Comment:
i don't like the use of `NgClass`. It uses `ngDoCheck` under the hook which executes at each change detection. 

You should do a normal binding.